tutorials/03-astropy_dev_env.rst

```diff
@@ -0,0 +1,159 @@
+   *astropy-dev* and activate it.
+
+   ::
+
+       conda create -n astropy-dev python=3 anaconda
+       source activate astropy-dev
+
+-  Inside the astropy-dev folder, install the Astropy from the source
+   code:
+
+   ::
+
+       python setup.py build
+       python setup.py install
+
+-  At this point, you could install any other packages you normally work
+   with, *except* for astropy. For example,
+
+   ::
+
+       conda install pandas
+
+Step 4: Make a new branch for editing the source code
+-----------------------------------------------------
+
+-  Inside the astropy-dev folder, create a new branch to work in. Try to
+   pick a descriptive name for your branch:
+
+   ::
+
+       git branch my-code-update
+       git checkout my-code-update
+
+Congratulations! You're ready to edit the astropy source code safely.
+Use ``git`` for committing your changes and pushing those changes to
+your ``my-code-update`` branch on Github.
+
+For more help with learning Git and Github, see the `Github
+Help <https://help.github.com/>`__ pages, the `Git and Github Learning
+Resources <https://help.github.com/articles/git-and-github-learning-resources/>`__
+guide, `try.github.io <http://try.github.io/>`_, and the `Github Learning Lab  <https://lab.github.com/>`_
+
+When you are done working on development, don't forget to exit the
+astropy-dev environment.
+
+::
+
+    source deactivate
+
+Keeping your fork up to date
+============================
+
+If it has been awhile since you last forked the Astropy repo, your fork
+will likely need updating. This tutorial will teach you how to use the
+command line to connect to the main Astropy Github repo, update to the
+latest development version, and push those changes to your personal
+Astropy fork on Github.
+
+Step 1: Add the core Astropy repo to your git config file
+---------------------------------------------------------
+
+-  Go to https://github.com/astropy/astropy and click on the green
+   "Clone or Download" button in the upper right-hand corner of the
+   page.
+-  Go to the directory where you have the Astropy repo on your computer,
+   for example:
+
+   ::
+
+       cd ~/astropy-dev
+
+-  Add a new remote named *upstream* to your local copy of the Astropy
+   repo
+
+   ::
+
+       git remote add upstream git@github.com:astropy/astropy.git
+
+Step 2: Pull any changes from the master branch of the main Astropy repo
+------------------------------------------------------------------------
+
+-  Make sure you are in the master branch of your local Astropy repo
+
+   ::
+
+       git checkout master
+
+-  Use ``git pull`` to update your local master branch from the upstream
+   Astropy master branch
+
+   ::
+
+       git pull upstream master
+
+Step 3: Push the changes to your fork on Github
+-----------------------------------------------
+
+-  Use ``git push`` to update your Github fork of Astropy:
+
+   ::
+
+       git push origin master
```

Comment:
Add that they may need to do a force push if they have in the past made changes directly to master (which seems o be a pretty common occurrence).

Reply:
I added a note